src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -453,6 +464,18 @@ protected override int MakeSlot(BoundExpression node)
                     break;
             }
             return base.MakeSlot(node);
+
+            bool isTopLevelMethod(MethodSymbol method)
+            {
+                switch (method.MethodKind)
```

Comment:
>switch (method.MethodKind) [](start = 16, length = 26)

Consider checking if container is a named type instead. This is how binder does that. Also, this approach will be more robust to future changes in the language. If new flavor of non-top level method will be added, etc.